add-topic-repos

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+GITHUB_API=${GITHUB_API:-https://api.github.com}
+DEFAULT_ORGANIZATION=$1
+TOPIC=$2
+exclude_pattern="(protected|private)"
+
+RED='\033[0;31m'
+GRAY='\033[1;30m'
+NC='\033[0m' # No Color
+
+print_separator() {
+    echo -e "${GRAY}============================================================${NC}"
+}
+
+set_topic_repo() {
+    repo=$1
+    topic=$2
+
+    endpoint=$GITHUB_API/repos/$repo/topics
+    echo repo:  "$repo"
+    echo topic:  "$topic"
+    echo endpoint:  "$endpoint"
+    print_separator
+
+    # get previous topics
+    result=$(curl \
+        -X GET \
+        -H "Authorization: token ${GH_TOKEN}" \
+        -H "Accept: application/vnd.github.mercy-preview+json" \
+        -sL "$endpoint" 2> /dev/null)
+
+    data=$(echo $result |
+     jq -r --arg topic "$topic" '.names + [$topic]')
+
+    curl \
+    -X PUT \
+     -H "Authorization: token ${GH_TOKEN}" \
+    -H "Accept: application/vnd.github.mercy-preview+json" \
+    -sL $endpoint \
+    -d "{\"names\": $data}"
+
+    return 0
+}
+
+read_repos() {
+    page=$2
+    topic=$3
+    result=$(curl -H "Authorization: token ${GH_TOKEN}" -sL $1?page=${page})
+    repos=$(echo $result | jq -r '.[].full_name')
+    repos_length=$(echo ${repos} | tr -d '\n' | wc -m)
+    if [ $repos_length -eq 0 ]; then
+        # there is no more projects to check
+        return 0
+    fi
+
+    process_count=0
+    for repo in $repos; do
+        if [[ ! $repo =~ $exclude_pattern ]] ; then
+            set_topic_repo "$repo" "$topic" &
+        fi
+	process_count=$(($process_count+1))
```

Comment:
It has a mixed tabs/spaces here, coud be good to review this in all file.

Reply:
done thx